src/platform/Linux/PlatformManagerImpl.cpp

```diff
@@ -203,7 +203,6 @@ CHIP_ERROR PlatformManagerImpl::_InitChipStack()
     SetConfigurationMgr(&ConfigurationManagerImpl::GetDefaultInstance());
     SetDiagnosticDataProvider(&DiagnosticDataProviderImpl::GetDefaultInstance());
     SetDeviceInfoProvider(&DeviceInfoProviderImpl::GetDefaultInstance());
```

Comment:
This should belong to Server, not be initialized here at all

Reply:
That is the final goal, we don't need to set those provider separately on each platform. Right now, not all platforms have implemented those instances yet, another solution is make GetDefaultInstance return nullptr by default on all platforms, and then we can set those provider in Server. I also need to check the initialization sequence of the server and make sure those providers are installed in time.

Created issue https://github.com/project-chip/connectedhomeip/issues/17400 to track this refactor.